packages/access-client/src/agent-data.js

```diff
@@ -17,6 +20,7 @@ export class AgentData {
   constructor(data, options = {}) {
     this.meta = data.meta
     this.principal = data.principal
+    this.sessionPrincipal = data.sessionPrincipal
```

Comment:
I think we should avoid storing a single `sessionPrincipal` on any of this agent stuff.
A single sessionPrincipal may make sense in a specific application context, but that application should keep track of it itself and e.g. pass it in to things like `addProvider`

Reply:
https://github.com/web3-storage/w3protocol/pull/546